File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for i in s:
            if len(stack) == 0:
                stack.append(i)
            else:
                if stack[-1] == '(' and i == ')':
                    stack.pop()
                elif stack[-1] == '{' and i == '}':
                    stack.pop()
                elif stack[-1] == '[' and i == ']':
                    stack.pop()
                else:
                    stack.append(i)
        if len(stack) != 0:
            return False
        return True

File: 0020-valid-parentheses/0020-valid-parentheses.kt
import java.util.*

class Solution {
    fun isValid(s: String): Boolean {
        val stack = Stack<String>()
        val splitted = s.split("").filter { !it.isEmpty() }
        for (i in splitted) {
            if (stack.isEmpty()) {
                stack.push(i)
                continue
            }
            when (i) {
                ")" -> if (stack.peek() == "(") stack.pop() else stack.push(i)
                "]" -> if (stack.peek() == "[") stack.pop() else stack.push(i)
                "}" -> if (stack.peek() == "{") stack.pop() else stack.push(i)
                else -> stack.push(i)
            }
        }
        return stack.isEmpty()
    }
}